### ISSUE TITLE
Castle.Core 4.1.2 Release? (Castle.Core 4.2.0 and 4.2.1 at AssemblyVersion 4.0.0)

### ISSUE THREAD

**Issue by kcamp** [castleproject/Core#322]
I am seeing an unexpected behavior at runtime.  The error itself makes sense and would typically be resolved by an AssemblyBindingRedirect.

```
Type: System.IO.FileLoadException
Message: Could not load file or assembly 'Castle.Core, Version=4.1.0.0, Culture=neutral, PublicKeyToken=407dd0808d44fbdc' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
```

However, looking at our _highest_ Castle.Core reference (4.2.1) for the correct redirect, I was surprised to see that 4.2.0 and 4.2.1 are reporting as `AssemblyVersion("4.0.0")`

Reading back through the [changelog ](https://github.com/castleproject/Core/blob/master/CHANGELOG.md)for 4.2.0, I see this came from GH-288.
The rationale + discussion in that thread make sense in terms of "why", particularly with regard to nuget and consumers of Castle.Core.  

Will there actually be 4.1.2 published as indicated to normalize the 4.1.x line to assembly version 4.0.0?  Or should we plan on updating all of our packages to 4.2.x?
> We'll definitely release 4.1.2 (with 4.0.0 assembly version) to fix that line

Thanks.

**Comment by fir3pho3nixx** [comment]
This was intentional, please see: https://github.com/castleproject/Core/issues/307

**Comment by fir3pho3nixx** [comment]
> Will there actually be 4.1.2 published as indicated to normalize the 4.1.x line to assembly version 4.0.0? Or should we plan on updating all of our packages to 4.2.x?

You are probably better upgrading.

**Comment by kcamp** [comment]
Acknowledged, thanks for the reply.

**Comment by jonorossi** [comment]
> Will there actually be 4.1.2 published as indicated to normalize the 4.1.x line to assembly version 4.0.0? Or should we plan on updating all of our packages to 4.2.x?

No, I decided after we had problems with NuGet doing annoying dependency resolution that it would just make things worse to release as it would likely break more things. As @fir3pho3nixx mentioned, it would be best to get on to 4.2.0 and away from those first few 4.x releases if you want to avoid binding redirects.
